crates/unleash-edge/src/offline_builder.rs

```diff
@@ -103,3 +113,40 @@ pub fn build_offline(offline_args: OfflineArgs) -> EdgeResult<CacheContainer> {
         Err(EdgeError::NoFeaturesFile)
     }
 }
+
+pub async fn build_offline_app_state(
+    args: CliArgs,
+    offline_args: OfflineArgs,
+) -> EdgeResult<(AppState, Vec<BackgroundTask>)> {
+    let (token_cache, features_cache, _, engine_cache) = build_offline(offline_args.clone())?;
+    let metrics_cache = Arc::new(MetricsCache::default());
+
+    let instance_data_sender = Arc::new(InstanceDataSending::SendNothing);
+
+    let app_state = AppState::builder()
+        .with_token_cache(token_cache.clone())
+        .with_features_cache(features_cache.clone())
+        .with_engine_cache(engine_cache.clone())
+        .with_metrics_cache(metrics_cache.clone())
+        .with_deny_list(args.http.deny_list.unwrap_or_default())
+        .with_allow_list(args.http.allow_list.unwrap_or_default())
+        .with_instance_sending(instance_data_sender)
+        .build();
+
+    let background_tasks =
+        create_offline_background_tasks(features_cache, engine_cache, offline_args);
+
+    Ok((app_state, background_tasks))
+}
+
+fn create_offline_background_tasks(
+    features_cache: Arc<FeatureCache>,
+    engine_cache: Arc<DashMap<String, EngineState>>,
+    offline_args: OfflineArgs,
+) -> Vec<Pin<Box<dyn Future<Output = ()> + Send>>> {
```

Comment:
Same thing here, since you went to the trouble of creating the type alias, why not just
```suggestion
) -> Vec<BackgroundTask> {
```